### PR TITLE
chore(ts): Convert ErrorRobot

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: src/sentry/static/sentry/app/components/errorRobot.tsx [hunk 1]
```diff
@@ -89,13 +92,12 @@ class ErrorRobot extends React.Component {
             project={project}
             source="issues_list"
             disabled={!project}
-            title={!project ? t('Select a project to create a sample event') : null}
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/errorRobot.tsx]: Does `CreateSampleEventButton` explicitly check for `null`?

**Review by billyvg — APPROVED** [review]

File: src/sentry/static/sentry/app/components/errorRobot.tsx (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/errorRobot.tsx]: It does not

File: src/sentry/static/sentry/app/views/issueList/noGroupsHandler/index.tsx [hunk 2]
```diff
@@ -85,15 +85,16 @@ class NoGroupsHandler extends React.Component<Props, State> {
...
+    const project = projects && projects.length > 0 ? projects[0] : undefined;
+    const sampleIssueId = groupIds.length > 0 ? groupIds[0] : undefined;
 
-    const sampleIssueId = this.props.groupIds.length > 0 ? this.props.groupIds[0] : '';
     return (
       <ErrorRobot
         org={organization}
         project={project}
-        sampleIssueId={sampleIssueId}
+        sampleIssueId={sampleIssueId?.toString() ?? ''}
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/issueList/noGroupsHandler/index.tsx]: Had to finess things here a bit. Ensures if the sample ID isn't available we set to `''` to indicate there is no first event.

File: src/sentry/static/sentry/app/views/onboarding/createSampleEventButton.tsx [hunk 3]
```diff
@@ -82,6 +82,10 @@ class CreateSampleEventButton extends React.Component<Props, State> {
     const {api, organization, project, source} = this.props;
     let eventData;
 
+    if (!project) {
+      return;
+    }
+
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/onboarding/createSampleEventButton.tsx]: `CreateSampleEvent` already supported being disabled if no project was specified.

File: src/sentry/static/sentry/app/components/errorRobot.tsx [hunk 4]
```diff
@@ -1,79 +1,82 @@
...
     const {sampleIssueId} = this.state;
 
     if (!project) {
       return;
     }
 
-    if (sampleIssueId === null || sampleIssueId === undefined) {
-      const url = '/projects/' + org.slug + '/' + project.slug + '/issues/';
-      const requestParams = {limit: 1};
+    if (sampleIssueId !== null && sampleIssueId !== undefined) {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/errorRobot.tsx]: maybe here you could use `defined(sampleIssueId)` . You have to import `import {defined} from 'app/utils'`

File: src/sentry/static/sentry/app/components/errorRobot.tsx [hunk 5]
```diff
@@ -1,79 +1,82 @@
...
-            error,
-            loading: false,
-          });
-        },
+        data: {limit: 1},
       });
+      this.setState({sampleIssueId: (data.length > 0 && data[0].id) || ''});
+    } catch (err) {
+      let error = err.responseJSON || true;
+      error = error.detail || true;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/errorRobot.tsx]:
```suggestion
      const error = err.responseJSON?.detail || true;
```

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/errorRobot.tsx]: Let's go all the way and use `??` then too :)

**Review by priscilawebdev — APPROVED** [review]
It's better now.. Good job! 👌

**Pull request merged**